src/dmd/s2ir.d

```diff
@@ -742,7 +742,8 @@ private extern (C++) class S2irVisitor : Visitor
         //printf("ExpStatement.toIR(), exp: %p %s\n", s.exp, s.exp ? s.exp.toChars() : "");
         if (s.exp)
         {
-            if (s.exp.hasCode)
+            if (s.exp.hasCode &&
+                s.exp.type != Type.tnoreturn) // e.g `assert(0)` not meant to be covered
```

Comment:
Unfortunately, this test picks up more than assert(0), it picks up any function that does not return. The user might be wanting to see if those noreturn functions get executed.

Replace with a check for HaltExp.

Reply:
I'd say that `Type.tnoreturn && e.op == TOK.assert_` would catch it, right ? let's say if at some point assert(0) codegen changes.

Reply:
Let's stick with `HaltExp`. It's highly unlikely that the codegen changes (it has been this way for ages), and if it does, the test will catch it.

Reply:
Yeah I tried but I forgot to mention that I've forced-pushed with  `HaltExp || AssertExp`  because both a required, otherwise cov was only of 66%.

Reply:
A line can marked more than once.